### ISSUE TITLE
FastAPI Parametric Dependency Injection

### ISSUE THREAD

**Issue by csuriano23** [strawberry-graphql/strawberry#2183]
Hello, is there a way to define a parametric dependency injection based on some of the query/mutation variables?

Eg. something like

```
def custom_context_dependency(name: str) -> str:
    return name


async def get_context(
    custom_value=Depends(custom_context_dependency),
):
    return {
        "custom_value": custom_value,
    }


@strawberry.type
class Query:
    @strawberry.mutation
    def example(self, name: str, info: Info) -> str:
        return f"Hello {info.context['custom_value']}"

```

**Comment by nrbnlulu** [comment]
You want to inject stuff into `info.context`?

**Comment by csuriano23** [comment]
Yes, I want to inject an http client as a dependency in the info.context, but some client parameters depends on the request parameter (eg. the login user)

**Comment by nrbnlulu** [comment]
You can:
- [write an extension](https://strawberry.rocks/docs/guides/custom-extensions#custom-extensions) and inject what you want there though extensions are not working for subscriptions currently.
-  pass `context_value` to `Schema.execute`

**Comment by patrick91** [comment]
maybe @Kludex has some ideas about this. In the django integration we have defined `get_context` on the class and it receives the info param, I think we should have the same for fastapi

**Comment by Kludex** [comment]
Why not having the HTTP client globally? 🤔

**Comment by Luferov** [comment]
This is a great idea. I would suggest writing an extension that injects the fastapi dependency into the resolvers.
